### PR TITLE
Optimize instance metadata availability check

### PR DESCRIPTION
**Pull request by atburke**
This PR optimizes the availability check for EC2 instance metadata, especially when Teleport isn't running on EC2.

### CODE REVIEW

File: lib/utils/ec2.go [hunk 1]
```diff
@@ -109,8 +111,21 @@ func NewInstanceMetadataClient(ctx context.Context) (*InstanceMetadataClient, er
...
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://169.254.169.254/latest/meta-data", nil)
+	if err != nil {
+		return false
+	}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
```
Comment (hatched) [inline lib/utils/ec2.go:127]: I think this should be under the actual call (ln 124) else it could return before it closes.

**Review by hatched — APPROVED** [review]
Looks good with some minor tweaks :)

File: lib/utils/ec2.go [hunk 2]
```diff
@@ -109,8 +111,21 @@ func NewInstanceMetadataClient(ctx context.Context) (*InstanceMetadataClient, er
 
 // IsAvailable checks if instance metadata is available.
 func (client *InstanceMetadataClient) IsAvailable(ctx context.Context) bool {
-	_, err := client.getMetadata(ctx, "")
-	return err == nil
+	// Doing this check via imds.Client.GetMetadata() involves several unrelated requests and takes a few seconds
+	// to complete when not on EC2. This approach is faster.
+	httpClient := http.Client{
+		Timeout: 250 * time.Millisecond,
```
Comment (r0mant) [inline lib/utils/ec2.go:117]: Is 250ms enough for metadata endpoint?

**Review by r0mant — APPROVED** [review]

File: lib/utils/ec2.go (hunk 1 above)
Comment (atburke) [reply lib/utils/ec2.go:131]: `Do` only returns a body that needs closing when err is nil, so I think this should be OK.

File: lib/utils/ec2.go (hunk 2 above)
Comment (atburke) [reply lib/utils/ec2.go:121]: In my testing, the whole check takes about 3ms when on EC2, but there's no harm in extending the timeout (other than the check taking longer) if that's what we want.

File: lib/utils/ec2.go (hunk 1 above)
Comment (hatched) [reply lib/utils/ec2.go:131]: Ahh ok thanks, good to know.

**Pull request merged**